examples/batch/batch.go

```diff
@@ -0,0 +1,146 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	pgx "github.com/jackc/pgx/v5"
+	pgxpool "github.com/jackc/pgx/v5/pgxpool"
+)
+
+type PgxIface interface {
+	Begin(context.Context) (pgx.Tx, error)
+	Close()
+}
+
+func databaseSetup(db PgxIface) (err error) {
+
+	// Initialize a database object
+	tx, err := db.Begin(context.Background())
+	if err != nil {
+		return fmt.Errorf("databaseSetup: %s", err)
+	}
+	// Finally, commit changes or rollback
+	defer func() {
+		switch err {
+		case nil:
+			err = tx.Commit(context.Background())
+		default:
+			_ = tx.Rollback(context.Background())
+		}
+	}()
+
+	// Create a new table 'ledger'
+	sql := `CREATE TABLE IF NOT EXISTS ledger (
+		id SERIAL PRIMARY KEY,
```

Comment:
it's better to use `bigserial` nowadays or even more standard compliant `bigint PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY`
Read [more](https://www.cybertec-postgresql.com/en/uuid-serial-or-identity-columns-for-postgresql-auto-generated-primary-keys/#should-i-use-integerserial-or-bigintbigserial-for-my-auto-generated-primary-key)